src/encoder/mod.rs

```diff
@@ -755,6 +758,27 @@ impl<'a, W: 'a + Write + Seek, T: ColorType> ImageEncoder<'a, W, T> {
         self.encoder.write_tag(Tag::YResolution, value).unwrap();
     }
 
+    /// Set image number of lines per strip
+    ///
+    /// This function must be called before calling write_data() or write_strip().
+    pub fn rows_per_strip(&mut self, value: u32) -> TiffResult<()> {
+        if self.strip_idx != 0 {
+            return Err(io::Error::new(
+                io::ErrorKind::InvalidInput,
+                "Cannot change strip size after data was written",
+            )
+            .into());
+        }
+        // Write tag as 32 bits
+        self.encoder.write_tag(Tag::RowsPerStrip, value)?;
```

Comment:
This tag is already written in `new`. Is that a problem?

Reply:
Calling `new_image` will directly construct the `ImageDecoder::new` and write the tags. Maybe it would be helpful to have an intermediate build structure, or another method with different argument type than the sizes.

Reply:
As far as I can see and test (see updated test), this is not a problem.
write_tag() does not write to file directly. It writes the tags into a map in memory. When a tag already exists  in the map, the value is overwritten with the new value. The map is written to file later when finish is called.